test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go

```diff
@@ -77,3 +85,82 @@ func ApplyProfile(profile *performancev2.PerformanceProfile) error {
 	}
 	return nil
 }
+
+// WaitForTuningUpdating is waiting for the cluster to transition into tuning configuration update state
+// done by PerformanceProfile application.
+func WaitForTuningUpdating(ctx context.Context, profile *performancev2.PerformanceProfile) {
+	GinkgoHelper()
+	// In case we are on OCP, we can query the MCP to determine if the update has started.
+	if !hypershift.IsHypershiftCluster() {
+		performanceMCP, err := mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+		testlog.Info("waiting for MCP starting to update")
+		mcps.WaitForCondition(performanceMCP, mcv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+		return
+	}
+
+	// On hypershift we can check the nodepool UpdatingConfig condition to determine if the update has started.
+	hostedClusterName, err := hypershift.GetHostedClusterName()
+	Expect(err).ToNot(HaveOccurred())
+	np, err := nodepools.GetByClusterName(ctx, testclient.ControlPlaneClient, hostedClusterName)
+	Expect(err).ToNot(HaveOccurred())
+	testlog.Infof("wait for node pool %q transition into update config state", client.ObjectKeyFromObject(np).String())
+	err = nodepools.WaitForUpdatingConfig(ctx, testclient.ControlPlaneClient, np.Name, np.Namespace)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// WaitForTuningUpdated is waiting for the cluster to come back from tuning configuration
+// done by PerformanceProfile application.
+// This is a lower lever function and should not be used directly.
+// Use PostUpdateSync instead
+func WaitForTuningUpdated(ctx context.Context, profile *performancev2.PerformanceProfile) {
+	GinkgoHelper()
+	// In case we are on OCP, we can query the MCP to determine if the update has completed.
+	if !hypershift.IsHypershiftCluster() {
+		performanceMCP, err := mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+		testlog.Infof("waiting for MCP being updated")
+		mcps.WaitForCondition(performanceMCP, mcv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		return
+	}
+
+	// On hypershift, we can check the nodepool UpdatingConfig condition to determine if the update has completed.
+	hostedClusterName, err := hypershift.GetHostedClusterName()
+	Expect(err).ToNot(HaveOccurred())
+	np, err := nodepools.GetByClusterName(ctx, testclient.ControlPlaneClient, hostedClusterName)
+	Expect(err).ToNot(HaveOccurred())
+	testlog.Infof("wait for node pool %q transition into config ready state", client.ObjectKeyFromObject(np).String())
+	err = nodepools.WaitForConfigToBeReady(ctx, testclient.ControlPlaneClient, np.Name, np.Namespace)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func EnforceLabels() {
```

Comment:
maybe a comment if this is a public function?  what are we enforcing here ?

Reply:
Done. Thanks!